### PR TITLE
Use release_ips() on _cleanup()

### PR DESCRIPTION
**Pull request by spikecurtis**
Before, we used unassign_ips() which was deprecated and
then removed in the latest `libcalico`.

### CODE REVIEW

**Comment by spikecurtis** [comment]
@metaswitchbot please test

File: calico_mesos/calico_mesos.py [hunk 1]
```diff
@@ -353,18 +353,10 @@ def _cleanup(hostname, container_id):
...
-        pools = datastore.get_ip_pools(ip.version)
-        for pool in pools:
-            if ip in pool:
-                # Ignore failure to unassign address, since we're not
-                # enforcing assignments strictly in datastore.py.
-                _log.info("Un-allocate IP %s from pool %s", ip, pool)
-                datastore.unassign_address(pool, ip)
+    # Release IP addresses.
+    ips = {net.ip for net in endpoint.ipv4_nets | endpoint.ipv6_nets}
+    _log.info("%s | Release IPs %s", container_id, ips)
```
Comment (djosborne) [inline calico_mesos/calico_mesos.py]: Why the container_id in the log? The line before it (in the logs) already details the container-id:
```
2016-01-07 00:22:51,113 2225 [INFO] CALICOMESOS 347: Cleaning executor with Container ID 8a77c818-55c5-4fe9-82b6-2d6a3fcf4425.
2016-01-07 00:22:51,114 2225 [INFO] CALICOMESOS 358: 8a77c818-55c5-4fe9-82b6-2d6a3fcf4425 | Release IPs set([IPAddress('192.168.0.23')])
```

Comment (spikecurtis) [inline calico_mesos/calico_mesos.py]: Because multiple of these processes can be running at once, the lines can get separated with other unrelated stuff.

I've added the PID now, so you can reconstruct a single call using that.  But, it might also be helpful for correlating to Mesos diags.
